crates/bevy_render/src/view/window.rs

```diff
@@ -184,15 +184,25 @@ pub fn prepare_windows(
             .entry(window.entity)
             .or_insert_with(|| unsafe {
                 // NOTE: On some OSes this MUST be called from the main thread.
-                let surface = render_instance.create_surface(&window.handle.get_handle());
-                let format = *surface
-                    .get_supported_formats(&render_adapter)
-                    .get(0)
-                    .unwrap_or_else(|| {
-                        panic!(
-                            "No supported formats found for surface {surface:?} on adapter {render_adapter:?}"
-                        )
-                    });
+                // As of wgpu 0.15, only failable if the given window is a HTML canvas and obtaining a WebGPU or WebGL 2 context fails.
+                let surface = render_instance
+                    .create_surface(&window.handle.get_handle())
+                    .unwrap();
+                let caps = surface.get_capabilities(&render_adapter);
+                let formats = caps.formats;
+                // Explicitally request an sRBG format, otherwise fallback to the first available format.
+                // For future HDR output support, we'll need to request a format that supports HDR,
+                // but as of wgpu 0.15 that is not yet supported.
+                let format = if formats.contains(&TextureFormat::Rgba8UnormSrgb) {
+                    TextureFormat::Rgba8UnormSrgb
+                } else if formats.contains(&TextureFormat::Bgra8UnormSrgb) {
+                    TextureFormat::Bgra8UnormSrgb
+                } else {
+                    // TODO: If this isn't an sRGB surface (eg. on webgpu), use an sRGB view_format on the SurfaceConfiguration.
+                    // TODO: webgpu doesn't support sRGB surfaces, and requires an sRGB view_format, but that was causing issues on Vulkan
+                    // so it's getting skipped for now.
+                    formats[0]
```

Comment:
Could we instead try to find the first sRGB format supported, else fall back to format[0]?

Reply:
Wouldn't that require iterating through all the formats available and checking if they're either `Bgra8UnormSrgb` or `Rgba8UnormSrgb`, and is IMO uglier than what's already there.

```rust
let good_formats = [TextureFormat::Bgra8UnormSrgb, TextureFormat::Rgba8UnormSrgb];
let mut format = formats[0];
for available_format in formats {
    if good_formats.contains(&available_format) {
        format = available_format;
        break;
    }
}
```

Reply:
I think iterate the formats and check `texture_format.describe().srgb`.

Reply:
I disagree with doing the .describe().srgb, as like rob says there's multiple formats that we don't want for a surface that are srgb (and in the future afaik we don't want an srgb surface for hdr output). Those formats should be filtered out by wgpu, but I'd prefer to explicitly request formats here as imo it's clearer (and there's only 2 that we want).

Reply:
I'm wondering if the preferred formats (i.e. those closer to the beginning of the list) are more likely to support more features like more MSAA modes or so. Also, my understanding is that some devices prefer Bgra over Rgba, but will support both. I don't know if there's a performance hit somewhere just for doing the swizzling. I think the hard-coded way overrides the preferences of the device, whereas the respecting-preferences way 'risks' choosing a 'poor' sRGB format that is less preferred and less suitable.

@cwfitzgerald - do you know if it is better to:
1. hard-code and request `Rgba8UnormSrgb`, else `Bgra8UnormSrgb`, else fall back to the first format in the list
2. or if it is better to look for the first sRGB format in the list, else fall back to the first format in the list
3. or just take the first format in the list and then manually apply sRGB in a shader if the format is not an sRGB format?

Reply:
I had missed that the PR was updated to default to the first format, and then iterate the list to look for either `Rgba8UnormSrgb` or `Bgra8UnormSrgb` and select the first encountered of those two if available. So it's a bit like option 2 above but filtering on the pool of allowed sRGB formats. Seems ok for now but I'm still interested in cwfitzgerald's input.